### ISSUE TITLE
:odd and :even not being refreshed

### ISSUE THREAD

**Issue by bh213** [ncannasse/domkit#36]
Can domkit components be dynamically added or removed to the scene, e.g. as a result of button click  and have CSS properly applied? 

I also noticed that :first-child & :last-child don't work with such dynamically added components (=outside component constructor) ..

I'd like to add proper example of how to add and remove components if there exists an "official way". Thanks.

See https://github.com/bh213/heaps-domkit-sample

**Comment by ncannasse** [comment]
Yes you can add and remove components dynamically. All you need it to call
style.sync() in your update loop.

Le sam. 29 janv. 2022 à 10:41, Gorazd Breskvar ***@***.***> a
écrit :

**Comment by bh213** [comment]
I think I got everything working correctly (even adding subtrees of components) but :odd and :even don't seem to update correctly when I remove e.g. first component from the list of the children.

Example here: https://bh213.github.io/heaps-domkit-sample/ -page 6, just click remove button

**Comment by ncannasse** [comment]
Oh i think indeed these cases are not handled ! Atm :odd/:even won't react
dynamically. That's indeed a bug, but not easy to fix at all.

Le dim. 30 janv. 2022 à 19:43, Gorazd Breskvar ***@***.***> a
écrit :
